library/swift/HeadersBuilder.swift

```diff
@@ -3,14 +3,19 @@ import Foundation
 private let kRestrictedPrefixes = [":", "x-envoy-mobile"]
 
 private func isRestrictedHeader(name: String) -> Bool {
-  return name == "host" || kRestrictedPrefixes.contains { name.hasPrefix($0) }
+  let isHostHeader = name.range(of: "host", options: [.caseInsensitive]) != nil
```

Comment:
This will match `FOOHOSTBAR`. This should work:

```suggestion
  let isHostHeader = name.caseInsensitiveCompare("host") == .orderedSame
```

Reply:
sorry, that's a pretty high level of sloppiness from my side. Updating.